resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/hadoopsteps/HadoopBootstrapUtil.scala

```diff
@@ -0,0 +1,186 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.k8s.features.hadoopsteps
+
+import java.io.File
+
+import scala.collection.JavaConverters._
+
+import com.google.common.base.Charsets
+import com.google.common.io.Files
+import io.fabric8.kubernetes.api.model.{ConfigMap, ConfigMapBuilder, ContainerBuilder, KeyToPathBuilder, PodBuilder}
+
+import org.apache.spark.deploy.k8s.Constants._
+import org.apache.spark.deploy.k8s.SparkPod
+import org.apache.spark.deploy.k8s.security.KubernetesHadoopDelegationTokenManager
+
+private[spark] object HadoopBootstrapUtil {
+
+   /**
+    * Mounting the DT secret for both the Driver and the executors
+    *
+    * @param dtSecretName Name of the secret that stores the Delegation Token
+    * @param dtSecretItemKey Name of the Item Key storing the Delegation Token
+    * @param userName Name of the SparkUser to set SPARK_USER
+    * @param fileLocation Location of the krb5 file
+    * @param krb5ConfName Name of the ConfigMap for Krb5
+    * @param pod Input pod to be appended to
+    * @return a modified SparkPod
+    */
+  def bootstrapKerberosPod(
+      dtSecretName: String,
+      dtSecretItemKey: String,
+      userName: String,
+      fileLocation: String,
+      krb5ConfName: String,
+      pod: SparkPod) : SparkPod = {
+      val krb5File = new File(fileLocation)
+      val fileStringPath = krb5File.toPath.getFileName.toString
+      val kerberizedPod = new PodBuilder(pod.pod)
+        .editOrNewSpec()
+          .addNewVolume()
+            .withName(SPARK_APP_HADOOP_SECRET_VOLUME_NAME)
+            .withNewSecret()
+              .withSecretName(dtSecretName)
+              .endSecret()
+            .endVolume()
+          .addNewVolume()
+            .withName(KRB_FILE_VOLUME)
+              .withNewConfigMap()
+                .withName(krb5ConfName)
+                .withItems(new KeyToPathBuilder()
+                  .withKey(fileStringPath)
+                  .withPath(fileStringPath)
+                  .build())
+                .endConfigMap()
+              .endVolume()
+        // TODO: (ifilonenko) make configurable PU(G)ID
+          .editOrNewSecurityContext()
+            .withRunAsUser(1000L)
+            .withFsGroup(2000L)
+            .endSecurityContext()
+          .endSpec()
+        .build()
+      val kerberizedContainer = new ContainerBuilder(pod.container)
+        .addNewVolumeMount()
+          .withName(SPARK_APP_HADOOP_SECRET_VOLUME_NAME)
+          .withMountPath(SPARK_APP_HADOOP_CREDENTIALS_BASE_DIR)
+          .endVolumeMount()
+        .addNewVolumeMount()
+          .withName(KRB_FILE_VOLUME)
+          .withMountPath(KRB_FILE_DIR_PATH)
```

Comment:
This makes the "/etc/" mounted as read only. I guess we should use as below:
          .withMountPath(KRB_FILE_DIR_PATH + "/krb5.conf")
          .withSubPath("krb5.conf")

Reply:
If it is read only why it cannot be read. Btw  if we plan to modify the configmap then there is a known limitation:  https://github.com/kubernetes/kubernetes/issues/50345.

Reply:
Hi Stavros,
The main aim of the following step was to mount the krb5.conf file into /etc/ directory.
        .addNewVolumeMount()
          .withName(KRB_FILE_VOLUME)
          .withMountPath(KRB_FILE_DIR_PATH) 
when the above is done all the contents which are actually  present inside the container in "/etc/" directory are lost and gets mounted only with the krb5.conf file with read permissions. As other contents are lost from "/etc" directory nothing works. As i had commented earlier as well, this makes the driver pod to fail (spawn fails) as well. To make things correct and mount only the krb5.conf file,   following should be done:
.addNewVolumeMount()
   .withMountPath(KRB_FILE_DIR_PATH + "/krb5.conf")
   .withSubPath("krb5.conf")

With this everything is perfect without any deletions from /etc/ directory. For krb5.conf file configmap, i dont think its going to change in run time. So this will not create any issue. Let me know, if you need more details on this. We can discuss further on this as well.

Regards
Surya